sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2118,6 +2129,26 @@ std::pair<RT::PiKernel, std::mutex *> ProgramManager::getOrCreateKernel(
                         &(BuildResult->MBuildResultMutex));
 }
 
+bool doesDevSupportImgAspects(const RTDeviceBinaryImage &Img,
+                              const device &Dev) {
+  const RTDeviceBinaryImage::PropertyRange &PropRange =
+      Img.getDeviceRequirements();
+  for (RTDeviceBinaryImage::PropertyRange::ConstIterator It : PropRange) {
+    using namespace std::literals;
+    if ((*It)->Name != "aspects"sv)
+      continue;
+    ByteArray Aspects = DeviceBinaryProperty(*It).asByteArray();
+    // Drop 8 bytes describing the size of the byte array.
+    Aspects.dropBytes(8);
+    while (!Aspects.empty()) {
+      aspect Aspect = Aspects.consume<aspect>();
```

Comment:
This might be widespread enough pattern to justify its own utility. Should go to a separate PR though.